### ISSUE TITLE
Emulator gets confused when multiple endpoints in a bot file are given the same id

### ISSUE THREAD

**Issue by johnataylor** [microsoft/BotFramework-Emulator#990]
## Version
Bot Framework Emulator (V4 PREVIEW) version: 4.0.0-preview.40025

## Describe the bug
If you have the same id for multiple endpoints in the bot file the emulator simply uses the last one in the array in the JSON when it makes calls however it displays the distinct endpoint uri and name.

Of course there is a user bug is in the bot file, however, the behavior of the tool makes this very cryptic.

## To Reproduce
1) Create a bot file with multiple endpoints, for example 3, but give each the same id.
2) Use the first or second from the displayed list.
3) Firstly you see multiple displayed in the left hand panel and secondly you see the endpoint uri corresponding to the name you picked, however, under the covers the emulator appears to use the last one added.

## Expected behavior
Either detected the error and done display the duplicates. Or detect the error in the file when you open it. Or change the data model such that the user provided id is not critical here.

**Comment by justinwilaby** [comment]
I have been in discussion with the team regarding BotFile validation and it seems we all agree this is the responsibility of the MSBot utility since all save/load/patch operations are piped through it. 

This is a case-and-point example of the rationale I presented when discussing this. The Emulator cannot know all the ways a BotFile can fail without a schema to validate against. We need a validation tool that prevents the user from opening an invalid file or we need to change the format of the BotFile to a binary one and only allow editing via the Emulator or MSBot.

**Comment by justinwilaby** [comment]
The `id` format Azure uses for *all* resources is a URI consisting of the following format `/subscription/<susbscriptionId>/resourceGroups/<resourceGroupId>/provider/<providerId>/<resourceType>/<userProvidedResourceName>` This guarantees unique ids for all resources.

**Comment by johnataylor** [comment]
Working with the latest download of the emulator version 4.2.1.

1) Start with the following bot file:

```
{
    "name": "asp-mvc-bot",
    "description": "",
    "services": [
      {
        "type": "endpoint",
        "name": "bot1 development",
        "endpoint": "http://localhost:3978/bot1",
        "appId": "",
        "appPassword": "",
        "id": "1"
      },
      {
        "type": "endpoint",
        "name": "bot2 development",
        "endpoint": "http://localhost:3978/bot2",
        "appId": "",
        "appPassword": "",
        "id": "2"
      },
      {
        "type": "endpoint",
        "name": "bot3 development",
        "endpoint": "http://localhost:3978/bot3",
        "appId": "",
        "appPassword": "",
        "id": "2"
      }
    ],
    "padlock": "",
    "version": "2.0"
}
```

Note this contains a bug. The id for the second and third entries in this file are the same.

2) Open this in the emulator.

You see three entries in the left hand side pane. The titles are correct and correspond to the entries in the bot file in the order given in the file. So:

![image](https://user-images.githubusercontent.com/2523898/51950019-c586b980-23e3-11e9-80b6-3e7041b59012.png)

3) Open tabs corresponding to each endpoint. You see the corresponding correct endpoint address in the view, for example:

![image](https://user-images.githubusercontent.com/2523898/51950135-14345380-23e4-11e9-9967-bf9561e3eb7b.png)

So for each endpoint in the left hand pane you see the correct endpoint address reflected in the chat window.

4) Now go ahead and use each bot.

In this particular case bot 3 is called when bot 2 is actually displayed. For example (and the bots here send an identifying number)...

![image](https://user-images.githubusercontent.com/2523898/51950220-7e4cf880-23e4-11e9-8d35-b037f6df91d0.png)


So this behavior still repros on the latest bits. But this is caused by having a bad bot file. However, the bot file is JSON and can be edited by hand. The resulting behavior in the emulator is very confusing because its shows one endpoint but calls another.

Expected behavior would be to show an error - or refuse to load the duplicate entry from the bot file.

**Comment by tonyanziano** [comment]
Closed in favor of tracking at #1611
